src/conn.rs

```diff
@@ -621,13 +624,16 @@ impl<const N: usize, P: ConnectionPeer> Connection<N, P> {
                     } else {
                         let seq = *syn;
                         *attempts += 1;
+
+                        // Re-trigger timeout if packet is still unacked, with exponential backoff.
```

Comment:
to me, "re-trigger timeout" means something like we would call `on_timeout` again, not that we are about to re-send the packet. I think the following is more clear:

```suggestion
                        // Double previous timeout for exponential backoff on each successive attempt.
```